manager/state/store/networks.go

```diff
@@ -183,6 +183,22 @@ func FindNetworks(tx ReadTx, by By) ([]*api.Network, error) {
 	return networkList, err
 }
 
+// FindNetwork is a utility function which returns the first
+// network for which the target string matches the ID, or
+// the name or the ID prefix.
+func FindNetwork(tx ReadTx, target string) *api.Network {
```

Comment:
No, it is a best effort function on the same line of the existing `Getnetwork()`.
If one query fails (ID, name or id pefix), it moves to the next one.

Eventually, the outcome will be the same as we would have if we reported an error happening during one query: We would eventually not be able to identify an network.

From the caller perspective, it will be a failure regardless.